Kudu.Tests/Core/Function/KafkaTriggerKedaAuthProviderTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Kudu.Core.Functions;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Kudu.Tests.Core.Function
+{
+    public class KafkaTriggerKedaAuthProviderTests
+    {
+        [Fact]
+        public void TestPopulateAuthenticationRef_Is_BindingCaseInsensitive()
+        {
+            KafkaTriggerKedaAuthProviderOverload kafkaTriggerKedaAuthProvider = new KafkaTriggerKedaAuthProviderOverload();
+            string jsonText = @"
+            {
+                ""Protocol"": ""SaslSsl"",
+                ""authenticationMode"": ""Plain"",
+                ""username"": ""test"",
+                ""password"": ""test""
+            }";
+
+            JToken jsonObj = JToken.Parse(jsonText);
+            IDictionary<string, string> authRef = kafkaTriggerKedaAuthProvider.PopulateAuthenticationRef(jsonObj, "testFunctionName");
+            Assert.Equal(1, authRef.Count);
+        }
+
+        [Fact]
+        public void PopulateAuthenticationRef_With_ProtocolData()
```

Comment:
TestPopulateAuthenticationRef_Is_BindingCaseInsensitive() tests the functioning where binding parameters will be used in case insensitive way. In the test case above jsondata is given as 

string jsonText = @"
            {
                ""protocol"": ""SaslSsl"",
                ""authenticationMode"": ""Plain"",
                ""username"": ""test"",
                ""password"": ""test""
            }";

where as in TestPopulateAuthenticationRef_Is_BindingCaseInsensitive() jsondata is given as 

  string jsonText = @"
            {
                ""Protocol"": ""SaslSsl"",
                ""authenticationMode"": ""Plain"",
                ""username"": ""test"",
                ""password"": ""test""
            }";
(P capital in protocol)